packages/shared/pkg/limit/upload.go

```diff
@@ -0,0 +1,34 @@
+package limit
+
+import (
+	"time"
+
+	"github.com/launchdarkly/go-sdk-common/v3/ldcontext"
+	"go.uber.org/zap"
+
+	featureflags "github.com/e2b-dev/infra/packages/shared/pkg/feature-flags"
+)
+
+func (l *Limiter) UpdateUploadLimitSemaphore() {
+	flagCtx := ldcontext.NewBuilder(featureflags.GcloudConcurrentUploadLimit).Build()
+
+	ticker := time.NewTicker(5 * time.Second)
```

Comment:
If we scale wide enough, this might become an issue. I'm not sure what launch darkly's rate limits are for this API, but we may want to increase this number (1 minute?) to give us some more head room. At the least, it should be a constant.

Example: if the rate limit is 100 requests per 10 second window, then this 5 second interval means we can only spin up 50 orchestrators before we hit that rate limit.

Reply:
They bill you per connections and there shouldn't be any rate limits. We used this to roll metrics or proxying which was calculated per sandbox.
Source: https://launchdarkly.com/docs/api#rate-limiting-and-sdks